### PR TITLE
[DOC] Update CONTRIBUTING [skip ci]

### PR DESCRIPTION
**Pull request by emdupre**
Updates the contributing guidelines and linked documentation.

- [X] Update CONTRIBUTING to be more beginner friendly
- [x] Update [documentation on adding interfaces](http://nipype.readthedocs.io/en/latest/devel/interface_specs.html) to include Neurostars, current mailing list, etc.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,40 +1,143 @@
...
+
+* [![Bugs](https://img.shields.io/badge/-bugs-fc2929.svg)][link_bugs] *These issues point to problems in the project.*
+
+    If you find new a bug, please provide as much information as possible to recreate the error.
+    The [issue template][link_issue_template] will automatically populate any new issue you open, and contains information we've found to be helpful in addressing bug reports.
+    Please fill it out to the best of your ability!
+
+    If you experience the same bug as one already listed in an open issue, please add any additional information that you have as a comment.
+
+* [![Help Wanted](https://img.shields.io/badge/-help%20wanted-c2e0c6.svg)][link_helpwanted] *These issues contain a task that a member of the team has determined we need additional help with.*
```
Comment (effigies) [inline CONTRIBUTING.md]: This links to "help wanted", not "help-wanted".

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,40 +1,143 @@
...
+
+    If you feel that you can contribute to one of these issues, we especially encourage you to do so!
+    Issues that are also labelled as [good-first-issue][link_good_first_issue] are a great place to start if you're looking to make your first contribution.
+
+* [![Enhancement](https://img.shields.io/badge/-enhancement-00FF09.svg)][link_enhancement] *These issues are asking for new features to be added to the project.*
```
Comment (effigies) [inline CONTRIBUTING.md]: This links to "feature", not "enhancement".

**Comment by emdupre** [comment]
Agh, thanks for catching those label links ! The commit I just pushed should fix both "help-wanted" and "enhancement" labels (successfully clicked through on my branch's copy).

File: CONTRIBUTING.md (hunk 1 above)
Comment (emdupre) [reply CONTRIBUTING.md]: 🤦‍♀️  Patched with b9a38d1 !

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,40 +1,143 @@
...
+**1. Comment on an existing issue or open a new issue referencing your addition.**
+
+This allows other members of the Nipype development team to confirm that you aren't overlapping with work that's currently underway and that everyone is on the same page with the goal of the work you're going to carry out.
+
+[This blog][link_pushpullblog] is a nice explanation of why putting this work in up front is so useful to everyone involved.
+
+**2. [Fork][link_fork] the [Nipype repository][link_nipype] to your profile.**
+
+This is now your own unique copy of Nipype.
+Changes here won't effect anyone else's work, so it's a safe space to explore edits to the code!
```
Comment (effigies) [inline CONTRIBUTING.md]: effect => affect

File: doc/devel/interface_specs.rst [hunk 4]
```diff
@@ -7,19 +7,18 @@ Interface Specifications
 Before you start
 ----------------
 
-Nipype is a young project maintained by an enthusiastic group of developers.
-Even though the documentation might be sparse or cryptic at times we strongly
-encourage you to contact us on the official nipype developers mailing list in
-case of any troubles: nipy-devel@neuroimaging.scipy.org (we are sharing a
-mailing list with the nipy community therefore please add ``[nipype]`` to the
-messsage title).
+Nipype is maintained by an enthusiastic group of developers, and we're excited to have you join!
```
Comment (effigies) [inline doc/devel/interface_specs.rst]: Might rephrase: "excited for you to join us!"

File: doc/devel/interface_specs.rst [hunk 5]
```diff
@@ -7,19 +7,18 @@ Interface Specifications
...
+In case of trouble, we encourage you to post on `NeuroStars <https://neurostars.org>`_ with the `nipype` tag.
+NeuroStars.org is a platform similar to StackOverflow but dedicated to neuroinformatics.
+You can also post on the nipype developers mailing list: http://mail.python.org/mailman/listinfo/neuroimaging.
+We are sharing a mailing list with the nipy community therefore please add ``[nipype]`` to the message title.
```
Comment (effigies) [inline doc/devel/interface_specs.rst]: This last sentence reads a little awkwardly to me. Perhaps:

"As we are sharing a mailing list with the nipy community, please add ..."

File: doc/devel/interface_specs.rst [hunk 6]
```diff
@@ -7,19 +7,18 @@ Interface Specifications
...
 
 
 Overview
 --------
 
-We're using the `Enthought Traits
-<http://code.enthought.com/projects/traits/>`_ package for all of our
+We're using the `Traits, formerly known as Enthought Traits
```
Comment (effigies) [inline doc/devel/interface_specs.rst]: I'd parenthesize the "formerly known as Enthought Traits" part, and not include it in the link text.

**Comment by effigies** [comment]
This LGTM. @mgxd if you want this in for the release, I don't see any complications.

**Comment by mgxd** [comment]
LGTM, thank you @emdupre

**Pull request merged**